app/src/api/index.ts

```diff
@@ -0,0 +1,204 @@
+     * @param data  ???
+     */
+    async addToLeaderboard(data: ApiAddToLeaderboardRequest): Promise<boolean> {
+        const response = await this.ax.put<string>('/leaderboard', data, {
+            responseType: 'text'
+        });
+        if (response.data === 'OK') {
+            return true;
+        }
+        return Promise.reject(response);
+    }
+
+    /**
+     * Запрос обновления пароля пользователя
+     * @param data  ???
+     * @return      Данные из таблицы рекордов
+     */
+    async getLeaderboard(data: ApiGetLeaderboardRequest): Promise<ApiGetLeaderboardResponse> {
+        const response = await this.ax.post<ApiGetLeaderboardResponse>('/leaderboard/all', data);
+        return response.data;
+    }
+
+    /**
+     * Перехватчик ответов от сервера. Позволяет выполнять обработку всех ответов,
+     * до возвращения в промис вызвавшего метода
+     * @param response  Объект ответ от сервера
+     */
+    private static responseHandler<T extends unknown>(response: AxiosResponse<T>) {
+        // eslint-disable-next-line no-console
+        console.log('[API]', response);
+        return response;
+    }
+
+    /**
+     * Перехватчик ошибок от сервера. Позволяет выполнять обработку всех ошибок,
+     * до возвращения в промис вызвавшего метода
+     * @param response  Объект ошибки от сервера
+     */
+    private static responseErrorHandler<T extends unknown>(responseError: AxiosError<T>) {
+        let badRequestError: ApiBadRequestError;
+
+        if (responseError.request.responseType === 'text' && typeof responseError.response.data === 'string') {
+            try {
+                badRequestError = JSON.parse(responseError.response.data);
+            } catch (error) {
+                // Несмотря на описание Api, формат ошибок иногда не такой как должен быть
+                badRequestError = {
+                    reason: responseError.response.data
+                };
+            }
+        } else {
+            badRequestError = responseError.response.data as ApiBadRequestError;
+        }
+
+        if (badRequestError) {
+            badRequestError.status = responseError.response.status;
+        }
+        // eslint-disable-next-line no-console
+        console.error('[API error]', badRequestError || responseError);
+
+        return Promise.reject(responseError);
+    }
+
+    /**
+     * Утилитный метод, изменяющий пути до аватаров на сервере с относительного адреса, на абсолютный.
+     * Необходим, т.к. пути возвращаемые сервером невозможно использвать в чистом виде
+     * @param objectWithAvatar  Объект содержащий аватар
+     */
+    private static getWithUpdatedAvatarPath<T extends ApiAvatarInfo>(objectWithAvatar: T) {
+        const newObjectWithAvatar = {...objectWithAvatar};
+        const {avatar} = newObjectWithAvatar;
+        if (avatar && avatar.startsWith('/')) {
+            newObjectWithAvatar.avatar = Api.url.origin + avatar;
+        }
+        return newObjectWithAvatar;
```

Comment:
```suggestion
        const {avatar} = objectWithAvatar;
 
        return {
          ...objectWithAvatar,
          avatar:  avatar?.startsWith('/') ? `${Api.url.origin}${newAvatar}` : avatar,
        }
```

Кроме того не очень очевидно, что в поле `avatar` находится путь до него. Я бы назвал поле `avatarUrl`

Reply:
Это поле приходит с сервера в таком виде. Не хотелось бы создавать новый объект только из за названия.